### PR TITLE
V3.2/task/flavor verification

### PR DESCRIPTION
**Pull request by isteffyx**
For this MR, we have included the verification implementation for new design.
We have excluded the esxi flow and unit test cases for this Merge request.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
+  https_proxy: "${HTTPS_PROXY}"
+  no_proxy: "${NO_PROXY}"
+
+stages:
+  - build
+  - package
+
+include:
+  - local: '/build/ci/build.yaml'
+  - local: '/build/ci/package-installer.yaml'
```
Comment (ddhawal) [inline .gitlab-ci.yml:17]: This file shouldn't have any changes

File: pkg/hvs/constants/verifier-rules-and-faults/constants.go [hunk 2]
```diff
@@ -1,62 +1,64 @@
...
+	FaultTagCertificateNotYetValid                  = FaultPrefix + "TagCertificateNotYetValid"
+	FaultXmlMeasurementLogContainsUnexpectedEntries = FaultPrefix + "XmlMeasurementLogContainsUnexpectedEntries"
+	FaultXmlMeasurementLogInvalid                   = FaultPrefix + "XmlMeasurementLogInvalid"
+	FaultXmlMeasurementLogMissing                   = FaultPrefix + "XmlMeasurementLogMissing"
+	FaultXmlMeasurementLogMissingExpectedEntries    = FaultPrefix + "XmlMeasurementLogMissingExpectedEntries"
+	FaultXmlMeasurementLogValueMismatchEntries384   = FaultPrefix + "XmlMeasurementLogValueMismatchEntriesSha384"
+	FaultXmlMeasurementsDigestValueMismatch         = FaultPrefix + "XmlMeasurementsDigestValueMismatch"
+	FaultXmlMeasurementValueMismatch                = FaultPrefix + "XmlMeasurementValueMismatch"
+	PcrEventLogUnexpectedFields                     = "PcrEventLogUnexpectedFields"
+	PcrEventLogMissingFields                        = "PcrEventLogMissingFields"
```
Comment (ddhawal) [inline pkg/hvs/constants/verifier-rules-and-faults/constants.go]: Fault/Policy Prefixes either be completely removed or added to new faults as well. I will prefer removing these prefixes.

File: pkg/lib/flavor/model/flavor.go [hunk 3]
```diff
@@ -1,90 +1,104 @@
...
+	// Hardware section is unique to Platform Flavor type
+	Hardware *Hardware                   `json:"hardware,omitempty"`
+	Pcrs     map[string]map[string]PcrEx `json:"pcrs,omitempty"`
+	// External section is unique to AssetTag Flavor type
+	External *External `json:"external,omitempty"`
+	Software *Software `json:"software,omitempty"`
+}
+
+//NewFVFlavor
+type NewFVFlavor struct {
```
Comment (ddhawal) [inline pkg/lib/flavor/model/flavor.go]: What does FV stand for?

File: pkg/lib/flavor/model/pcrex.go [hunk 4]
```diff
@@ -1,27 +1,63 @@
...
+// the PCR state from system boot
+type PcrEx struct {
+	Value string             `json:"value"`
+	Event []hcTypes.EventLog `json:"event,omitempty"`
+}
+
+type EventLogCreteria struct {
+	TypeID      string    `json:"type_id"`   //oneof-required
+	TypeName    string    `json:"type_name"` //oneof-required
+	Tags        *[]string `json:"tags,omitempty"`
```
Comment (ddhawal) [inline pkg/lib/flavor/model/pcrex.go]: avoid pointers to slice

File: pkg/lib/flavor/types/esx_platform_flavor.go [hunk 5]
```diff
@@ -1,353 +1,352 @@
...
+
+// getAssetTagFlavor returns the asset tag part of the flavor including the certificate and
+// all the key-value pairs that are part of the certificate.
+func (esxpf ESXPlatformFlavor) getAssetTagFlavor() ([]cm.Flavor, error) {
+	log.Trace("flavor/types/esx_platform_flavor:getAssetTagFlavor() Entering")
+	defer log.Trace("flavor/types/esx_platform_flavor:getAssetTagFlavor() Leaving")
+
+	var errorMessage = "Error during creation of ASSET_TAG flavor"
+	var err error
+	//var tagCertificateHash []byte
```
Comment (ddhawal) [inline pkg/lib/flavor/types/esx_platform_flavor.go]: why code has been commented out?

File: pkg/lib/host-connector/types/pcr_manifest.go [hunk 6]
```diff
@@ -1,385 +1,499 @@
...
+}
+
+// Finds the EventLogEntry in a PcrEventLogMap provided the pcrBank and index.  Returns
+// null if not found.  Returns an error if the pcrBank is not supported
+// by intel-secl (currently supports SHA1 and SHA256).
+func (pcrEventLogMap *NewFVPcrEventLogMap) GetEventLog(pcrBank SHAAlgorithm, pcrIndex int) (*NewFVEventLogEntry, error) {
+
+	var eventLogEntry *NewFVEventLogEntry
+
+	if pcrBank == SHA1 {
```
Comment (ddhawal) [inline pkg/lib/host-connector/types/pcr_manifest.go]: Use switch case instead

**Review by ddhawal — CHANGES_REQUESTED** [review]

File: pkg/lib/flavor/model/pcrex.go (hunk 4 above)
Comment (isteffyx) [reply pkg/lib/flavor/model/pcrex.go]: We have removed this pointer.

File: pkg/lib/flavor/model/flavor.go (hunk 3 above)
Comment (isteffyx) [reply pkg/lib/flavor/model/flavor.go]: Modified this 'NewFVFlavor' struct name to 'Flavor'.

File: pkg/lib/flavor/types/esx_platform_flavor.go (hunk 5 above)
Comment (isteffyx) [reply pkg/lib/flavor/types/esx_platform_flavor.go]: Removed this comment syntax.

File: pkg/hvs/constants/verifier-rules-and-faults/constants.go (hunk 2 above)
Comment (isteffyx) [reply pkg/hvs/constants/verifier-rules-and-faults/constants.go]: PcrEventLogUnexpectedFields and  PcrEventLogMissingFields variables are not used for faults statement. These are used to inform the user if there are any mismatched fields(like Tags, Type name) between expected and actual event log values while evaluating equals and includes rule.

File: .gitlab-ci.yml (hunk 1 above)
Comment (isteffyx) [reply .gitlab-ci.yml:17]: We have reverted the changes in yml file

**Pull request closed without merging**